packages/angular/cli/src/command-builder/command-module.ts

```diff
@@ -139,9 +139,17 @@ export abstract class CommandModule<T extends {} = {}> implements CommandModuleI
     }
 
     // Gather and report analytics.
+    let analyticsFlushPromise = Promise.resolve();
+    let analyticsFlushInterval: NodeJS.Timeout | undefined;
     const analytics = await this.getAnalytics();
+
     if (this.shouldReportAnalytics) {
       await this.reportAnalytics(camelCasedOptions);
+
+      // Flush on an interval (if the event loop is waiting).
+      analyticsFlushInterval = setInterval(() => {
+        analyticsFlushPromise = analyticsFlushPromise.then(() => analytics.flush());
+      }, 2000);
```

Comment:
I forget the exact edge cases of `setInterval()`, but is there a risk multiple tasks get queued and executed at the same time? What about if a report to analytics is very slow, or we're hung on a long synchronous task?

https://developer.mozilla.org/en-US/docs/Web/API/setInterval#ensure_that_execution_duration_is_shorter_than_interval_frequency

Reply:
That should not be a problem in this case as calling `flush` multiple times before the http request is done will not trigger the same events.

See: https://github.com/angular/angular-cli/blob/05a0b28b2f5b1a67d5c6cb6a1e1cce64c06e2329/packages/angular/cli/src/analytics/analytics-collector.ts#L158-L161